kong/plugins/rate-limiting/policies/init.lua

```diff
@@ -175,69 +249,96 @@ return {
       end
 
       return 0
-    end
+    end,
   },
   ["redis"] = {
     increment = function(conf, limits, identifier, current_timestamp, value)
-      local red, err = get_redis_connection(conf)
-      if not red then
-        return nil, err
-      end
-
-      local keys = {}
-      local expiration = {}
-      local idx = 0
       local periods = timestamp.get_timestamps(current_timestamp)
-      for period, period_date in pairs(periods) do
-        if limits[period] then
-          local cache_key = get_local_key(conf, identifier, period, period_date)
-          local exists, err = red:exists(cache_key)
-          if err then
-            kong.log.err("failed to query Redis: ", err)
-            return nil, err
-          end
 
-          idx = idx + 1
-          keys[idx] = cache_key
-          if not exists or exists == 0 then
-            expiration[idx] = EXPIRATION[period]
+      if conf.sync_rate <= 0 then
```

Comment:
ditto (inconsistency with rla): `sync_rate = 0` and `sync_rate = -1` would result in the same synchronous behaviour

Reply:
done